_source/_posts/2020-10-08-okta-spring-security-feature-flags.md

```diff
@@ -0,0 +1,442 @@
+
+The second parameter is the name of the treatment in Split that we want to target.
+
+The third parameter sends a map where the key is **groups** and the value is the list of group names that the authenticated user belongs to. In the case of our user **linda**, this will be **Everyone** (among some other defaults). In the case of our user **louise**, this will be **Everyone** and **BETA_TESTER**.
+
+The last line of the controller method now returns the **home-beta** template if the result of **getTreatment** is **on** and the **home** template otherwise.
+
+Fire up the app and try to login as **linda** in an incognito window. You should see the same home template as before. Kill that incognito window, open another one and login as **louise**. You should see the new beta template.
+
+{% img blog/okta-split-spring-security/beta-experience.png alt:"beta experience" width:"600" %}
+
+### Make Your New Functionality Generally Available
+
+Now that we have different treatments of our home template for regular users and beta testers, you may be wondering how to go about making the beta template available to everyone?
+
+Split makes it easy-peasy. Go back to your Split definition for **BETA_UI_EXPERIENCE** and switch the **serve** setting in the **Set The Default Rule** section from **off** to **on**. Save and Confirm the change.
+
+Without restarting the Spring Boot application, login as **linda** again. You should now see the same page that you saw for **louise** earlier.
+
+Pretty cool, eh?
+
+### How to Repeat the Beta / Release Cycle
+
+With this architecture in place, it's now very easy to set up a new beta cycle. The steps would be something like this:
+
+1. Copy the `home-beta.html` template to `home.html` (now that it's ready for production)
+2. Create a new `home-beta.html` template
+3. Set the default rule back to **off** in Split
+4. Redeploy the app
+5. Let your beta testers test the new experience
+6. When ready, set the default rule back to **on** in Split
+7. NO NEED TO REDEPLOY
+
+You could do this over and over again and never touch the controller code. The only thing that's changing are the templates and the settings in split.
+
+This approach also lends itself to changing who is in your beta test program without having to change your code.
+
+In a real application, you'd be working with a database or an Identity Management system where you could add and remove users from the **BETA_TESTER** group. With Okta, you can easily manage who belongs to the group from the admin console or via the Okta Management API. Those users would always see the latest and greatest beta while ordinary users would see only the current release.
+
+## Learn More About Building Secure Applications
+
+I hope you've seen how useful it can be to set up different experiences for different users using Okta, Split and the native functionality built into Spring Security.
+
```

Comment:
Is there a GitHub repo you can link to?

Reply:
yep. I pushed up a new version with that link.